gpytorch/lazy/block_lazy_tensor.py

```diff
@@ -60,6 +61,42 @@ def _expand_batch(self, batch_shape):
         res = self.__class__(self.base_lazy_tensor._expand_batch(batch_shape))
         return res
 
+    def _getitem(self, row_index, col_index, *batch_indices):
+        # First the easy case: just batch indexing
+        if _is_noop_index(row_index) and _is_noop_index(col_index):
+            return self.__class__(self.base_lazy_tensor._getitem(row_index, col_index, *batch_indices, _noop_index))
+
+        # If either of the dimensions are indices, it's too complicated - go with the base case
+        if not isinstance(row_index, slice) or not isinstance(col_index, slice):
+            # It's too complicated to deal with tensor indices in this case - we'll use the super method
+            return super()._getitem(row_index, col_index, *batch_indices)
```

Comment:
I assume so, but is it verified that the actual use cases leading to the 3 issues aren't punted on by this? Presumably it was just slice indexing.

Reply:
Ah, just saw the unit test 👍

Reply:
Actually - what we can do to solve for those use cases is to call `evaluate_kernel` before the super()._getitem (maybe while throwing a warning that we're doing this). This will get around all the corner cases.